### ISSUE TITLE
linalg.cross doesn't run checks on broadcasted dimensions

### ISSUE THREAD

**Issue by samdow** [pytorch/pytorch#83756]
### 🐛 Describe the bug

Two small examples, I would expect both to work based on the [docs](https://pytorch.org/docs/stable/generated/torch.linalg.cross.html#torch.linalg.cross) specifically saying that it will error
"If after broadcasting input.size(dim) != 3 or other.size(dim) != 3." I expected the inverse to be true that it will not error if after broadcasting, these are checks are met

```python
import torch
from torch import linalg

linalg.cross(torch.randn(2, 3), torch.randn(5, 2, 3), dim=-1) # Dimension -1 does not have size 3
linalg.cross(torch.randn(2, 3), torch.randn(5, 2, 3), dim=2)  # Dimension out of range
```

I wanted to check if either or both of these are intentional

### Versions

master

cc @jianyuh @nikitaved @pearu @mruberry @walterddr @IvanYashchuk @xwang233 @Lezcano

**Comment by Lezcano** [comment]
This is a bug. Next week I can fix this one, together with https://github.com/data-apis/array-api/issues/415#issuecomment-1101130081 and https://github.com/pytorch/pytorch/issues/77629. It shouldn't be too difficult
